lib/spack/spack/cmd/create.py

```diff
@@ -730,6 +746,20 @@ def get_url(args):
     return url
 
 
+def is_git_url(url):
+    """Check if the URL is likely to be a git repository. The code doesn't attempt
+    to clone the repository!
+
+    Args:
+        url (str): The url to check
+
+    Returns:
+        bool: True if it seems to be a git repository
+    """
+
+    return url.startswith('git://') or url.startswith('git@') or url.endswith('.git')
```

Comment:
What about https://github.com/foo/bar? If there are no other things tacked onto the end it would be valid. May need a regex to test this.

Reply:
As far as I understand, in general `.git` is required - Github forgives missing `.git` quietly, Gitlab does a redirect (and git warns about it). Some git servers could be less forgiving and just return an error.

Reply:
For our current packages, it looks like 22 out of 171 packages that use the `git` attribute don't append `.git`. We might want to make a special exception for GitHub.

Reply:
> For our current packages, it looks like 22 out of 171 packages that use the `git` attribute don't append `.git`. We might want to make a special exception for GitHub.

... or fix those 22 package. Or even leave them as-is, since the PR only changes `create` command, it doesn't affect any existing packages.

Reply:
@adamjstewart  After doing a few checks with different git servers (github, gitlab, whatever linux kernel git uses), I find that the `.git` suffix is indeed optional. I have now added `--git` switch to force using the `url` argument passed to `spack create` as a value for `git` attribute rather than `url` one. Let me know if you think removing the detection code and relying on that switch will be a better solution (and implying `--git` if `--branch`, `--tag` or `--commit` are set)